lib/internal/util/inspect.js

```diff
@@ -638,13 +638,23 @@ function formatRaw(ctx, value, recurseTimes) {
         return ctx.stylize(regExpToString(value), 'regexp');
       base = `${regExpToString(value)}`;
     } else if (isDate(value)) {
+      const prefix = getPrefix(constructor, '', 'Date');
+      let newVal;
+      if (Number.isNaN(dateGetTime(value))) {
+        newVal = Object.getPrototypeOf(value) ?
+          value : prefix + value;
+      } else {
+        newVal = Object.getPrototypeOf(value) ?
+          dateToISOString(value) : prefix + dateToISOString(value);
```

Comment:
Perhaps we could perform `dateToISOString(value)` just once, since it's used in both branches?